Projects/App/Sources/Journeys/ClaimsJourney.swift

```diff
@@ -86,15 +76,29 @@ extension AppJourney {
             style: .detented(.scrollViewContentSize, modally: false)
         ) { action in
             if case .didAcceptHonestyPledge = action {
-                ContinueJourney()
-                    .onPresent {
-                        @PresentableStore var store: ClaimsStore
-                        store.send(.startClaim(from: origin.id))
+                let status = UNUserNotificationCenter.current().startClaimIfStatusDeterminated(forOriginId: origin.id)
+                if case .notDetermined = status {
+                    Journey(
+                        ClaimsAskForPushnotifications(),
+                        style: .detented(.large, modally: false)
+                    ) { _ in
+                        PopJourney()
+                            .onPresent {
+                                let store: ClaimsStore = globalPresentableStoreContainer.get()
+                                store.send(.startClaim(from: origin.id))
+                            }
                     }
+                }
             } else {
                 ClaimJourneys.getScreenForAction(for: action, withHidesBack: true)
             }
         }
+        .onAction(ClaimsStore.self) { action, nav in
+            if case .startClaim = action {
+                nav.viewController.navigationController?.popToViewController(nav.viewController, animated: true)
```

Comment:
This seems odd to me, why did you have to do this?

Reply:
Removed!